File: solutions/elixir/gigasecond/1/lib/gigasecond.ex
defmodule Gigasecond do
  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) ::
          {{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}
  def from({{year, month, day}, {hours, minutes, seconds}}) do
    {:ok, moment} = NaiveDateTime.new(year, month, day, hours, minutes, seconds)
    NaiveDateTime.add(moment, 1_000_000_000, :second)
    |> NaiveDateTime.to_erl()
    
  end
end

File: solutions/go/gigasecond/1/gigasecond.go
// Package gigasecond provides operations to add a Gigaseconds to a time.
package gigasecond

import "time"

// AddGigasecond adds 1 Gigaseconds (1 billion seconds) to a time
// a returns the new time.
func AddGigasecond(t time.Time) time.Time {
	return t.Add(time.Second * 1_000_000_000)
}
